### ISSUE TITLE
`pp::/fail-fast?` behaviour is broken for mutations

### ISSUE THREAD

**Issue by bwalex** [wilkerlucio/pathom#69]
I just can't get an exception to actually be thrown when I throw it within a mutation. `::pp/fail-fast?` doesn't do anything, the explicit error handler doesn't do anything. It seems to me like the code in `pp/parser` *always* does try/catch for a mutation, and so I always get a `::pp/error` value instead of the exception being thrown.

The relevant code in `pathom.parser` is this, in particular the call `(action)` in the try-catch block.

```clojure
(defn parser [{:keys [read mutate]}]
  (fn self [env tx]
    (tracing env {::pt/event ::parse-loop}
      (let [{:keys [children] :as tx-ast} (or (::ast tx) (query->ast tx))
            tx  (vary-meta tx assoc ::ast tx-ast)
            env (-> env
                    (assoc :parser self))]
        (loop [res {}
               [{:keys [query key type params] :as ast} & tail] children]
          (if ast
            (let [_     (trace env {::pt/event ::process-key :key key})
                  query (cond-> query (vector? query) (vary-meta assoc ::ast tx-ast))
                  env   (cond-> (merge env {:ast ast :query query})
                          (nil? query) (dissoc :query)
                          (= '... query) (assoc :query tx))
                  value (case type
                          :call
                          (do
                            (assert mutate "Parse mutation attempted but no :mutate function supplied")
                            (let [{:keys [action]} (mutate env key params)]
                              (if action
                                (try
                                  (action)
                                  (catch #?(:clj Throwable :cljs :default) e
                                    {::error e})))))

                          (:prop :join :union)
                          (do
                            (assert read "Parse read attempted but no :read function supplied")
                            (read env))

                          nil)]
              (recur (assoc res (ast->out-key ast) value) tail))
            res))))))
```

**Comment by bwalex** [comment]
In the same spirit, it seems only the parallel-parser supports the `process-error` bit of `p/error-handler-plugin`.

**Comment by wilkerlucio** [comment]
Yeah, the error been captured at parser level is preventing all that, I did
a fix, just need to test a bit more and it will be out on the next release.

**Comment by wilkerlucio** [comment]
Closed by 8af662fc06c41de802c0cbe0d5395c6991d21700
